pkg/cmd/cli/cmd/newapp.go

```diff
@@ -543,7 +542,7 @@ func setAnnotations(annotations map[string]string, result *newcmd.AppResult) err
 	for _, object := range result.List.Items {
 		err := util.AddObjectAnnotations(object, annotations)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to add annotation to object of type %q, this type is probably unknown.", object.GetObjectKind().GroupVersionKind())
```

Comment:
An "unknown type" seems some what vague ... how about combine a flavor of what the error message used to be .... perhaps something like "..., this type does not implement the necessary interface(s) to allow the add annotation operation to succeed."